shared/js/notification_helper.js

```diff
@@ -30,26 +30,29 @@
     // an object -> {raw: string}
     send: function nh_send(titleL10n, options) {
       return new Promise(function(resolve, reject) {
-        navigator.mozL10n.once(function() {
-          var title = getL10n(titleL10n);
+        navigator.mozL10n.once(() => {
+          getL10n(titleL10n).then(title => {
+            options.title = title;
+            if (options.bodyL10n) {
```

Comment:
Looks that what you really want is `Promise.all([titleL10n, options.bodyL10n].map(getL10n))` instead of this nested spaghetti :)